<launch>
    <arg name="name" default="face_cropping"/>
    <arg name="output" default="log"/>

    <arg name="input_camera_topic" default="/camera1/image_raw"/>
    <arg name="output_camera_topic" default="/face_camera/image_raw"/>

    <arg name="refresh_rate" default="30.0"/>
    <arg name="width" default="400"/>
    <arg name="height" default="400"/>

    <arg name="frames_used_for_stabilizer" default="15"/>

    <arg name="min_width_change" default="0.2"/>
    <arg name="min_height_change" default="0.4"/>
    <arg name="min_x_change" default="0.1"/>
    <arg name="min_y_change" default="0.1"/>

    <arg name="right_margin" default="0.2"/>
    <arg name="left_margin" default="0.2"/>
    <arg name="top_margin" default="0.2"/>
    <arg name="bottom_margin" default="0.25"/>

    <arg name="is_peer_image" default="false"/>

    <arg name="dnn_model_path" default="$(find face_cropping)/models/res10_300x300_ssd_iter_140000.caffemodel"/>
    <arg name="dnn_deploy_path" default="$(find face_cropping)/models/deploy.prototxt"/>

    <arg name="use_gpu" default="false"/>

    <node name="$(arg name)" pkg="face_cropping" type="face_cropping" respawn="true" output="$(arg output)">
        <remap from="input_image" to="$(arg input_camera_topic)"/>
        <remap from="output_image" to="$(arg output_camera_topic)"/>

        <param name="name" type="string" value="$(arg name)"/>
        <param name="refresh_rate" type="double" value="$(arg refresh_rate)"/>
        <param name="width" type="int" value="$(arg width)"/>
        <param name="height" type="int" value="$(arg height)"/>

        <param name="frames_used_for_stabilizer" type="int" value="$(arg frames_used_for_stabilizer)"/>

        <param name="min_width_change" type="double" value="$(arg min_width_change)"/>
        <param name="min_height_change" type="double" value="$(arg min_height_change)"/>
        <param name="min_x_change" type="double" value="$(arg min_x_change)"/>
        <param name="min_y_change" type="double" value="$(arg min_y_change)"/>

        <param name="right_margin" type="double" value="$(arg right_margin)"/>
        <param name="left_margin" type="double" value="$(arg left_margin)"/>
        <param name="top_margin" type="double" value="$(arg top_margin)"/>
        <param name="bottom_margin" type="double" value="$(arg bottom_margin)"/>

        <param name="is_peer_image" type="bool" value="$(arg is_peer_image)"/>

        <param name="dnn_model_path" type="string" value="$(arg dnn_model_path)"/>
        <param name="dnn_deploy_path" type="string" value="$(arg dnn_deploy_path)"/>

        <param name="use_gpu" type="bool" value="$(arg use_gpu)"/>
    </node>
</launch>
